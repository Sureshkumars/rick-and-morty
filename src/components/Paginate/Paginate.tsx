import React from "react";
import style from "./index.module.css";

interface PaginateProps {
  currentPage: number;
  pageCount: number;
  onPageClick: (currentPage: number) => void;
}

const Paginate = (props: PaginateProps) => {
  const { currentPage, pageCount, onPageClick } = props;

  const isFirst = currentPage === 1;
  const isLast = currentPage === pageCount;

  const pages = Array.from(Array(pageCount).keys()).map((value) => value + 1);

  return (
    <div className={style.paginationContainer}>
      <div className={style.paginationButtonContainer}>
        {!isFirst && (
          <div
            className={style.pagination_button}
            onClick={() => onPageClick(currentPage - 1)}
          >
            &lt;
          </div>
        )}
        {pages.map((page) => {
          const disabled =
            page !== 1 && page !== pageCount
              ? Math.abs(page - currentPage) > 1
              : false;
          const isPreviousDisabled =
            page - 1 !== 1 && page - 1 !== pageCount
              ? Math.abs(page - 1 - currentPage) > 1
              : false;

          if (isPreviousDisabled && disabled) {
            return null;
          }

          return (
            <div
              key={page}
              className={`${style.pagination_button} ${
                page === currentPage ? style.highlight : ""
              } ${disabled ? style.disabled : ""}`}
              onClick={() => page !== currentPage && onPageClick(page)}
            >
              {disabled ? "..." : page}
            </div>
          );
        })}
        {!isLast && (
          <div
            className={style.pagination_button}
            onClick={() => onPageClick(currentPage + 1)}
          >
            &gt;
          </div>
        )}
      </div>
    </div>
  );
};

export default Paginate;
