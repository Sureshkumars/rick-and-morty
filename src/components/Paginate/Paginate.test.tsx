import React from "react";
import Paginate from "./Paginate";
import { render } from "@testing-library/react";

it("Paginate matches snapshot", () => {
  let currentPage: number = 1;
  const { asFragment }: any = render(
    <Paginate
      currentPage={currentPage}
      pageCount={10}
      onPageClick={(pageValue) => {
        //Clicked Page Value
        currentPage = pageValue;
      }}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
