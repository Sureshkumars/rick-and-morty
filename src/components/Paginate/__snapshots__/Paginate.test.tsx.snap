// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginate matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="paginationContainer"
  >
    <div
      class="paginationButtonContainer"
    >
      <div
        class="pagination_button highlight "
      >
        1
      </div>
      <div
        class="pagination_button  "
      >
        2
      </div>
      <div
        class="pagination_button  disabled"
      >
        ...
      </div>
      <div
        class="pagination_button  "
      >
        10
      </div>
      <div
        class="pagination_button"
      >
        &gt;
      </div>
    </div>
  </div>
</DocumentFragment>
`;
